src/python/Utils/ExecuteDAO.py

```diff
@@ -0,0 +1,214 @@
+    if logger.handlers:
+        logger.handlers.clear()
+    logger.addHandler(outHandler)
+    logger.setLevel(logLevel)
+    return logger
+
+
+class ExecuteDAO():
+    """
+    A generic class to create the DAO Factory and execute the DAO module.
+    """
+    def __init__(self, logger, package=None, daoModule=None):
+        """
+        __init__
+        The ExecuteDAO constructor method.
+        :param logger: The logger instance.
+        :param package: The Package from which the DAO factory to be initialised.
+        :param module: The DAO module to be executed.
+        """
+        # Connecting to DB
+        myThread = threading.currentThread()
+        connectToDB()
+        self.logger = logger
+        self.dbi = myThread.dbi
+        self.package = package
+        self.daoModule = daoModule
+
+        self.daoFactory = DAOFactory(package=package,
+                                     logger=self.logger,
+                                     dbinterface=self.dbi)
+        self.logger.info("DAO Factory initialised from package: %s", self.package)
+
+        self.dao = self.daoFactory(classname=daoModule)
+        self.logger.info("DAO Module initialised as: %s", self.daoModule)
+
+    def __call__(self, dryRun=False, *sqlArgs, **sqlKwArgs):
+        """
+        __call__
+        The ExecuteDAO call method. This is the method to forward all provided
+        arguments to the execute method of the DAO and return the result from the query
+        :param dryRun:      Bool flag to indicate dryrun method
+        :param *sqlArgs:    All positional arguments to be forwarded to the DAO's execute method.
+        :param **sqlKwArgs: All named arguments to be forwarded to the DAO's execute method.
+        :return:            The result from the DAO (could be either a dict or a list depending on the DAO itself)
+        """
+        if dryRun:
+            results = None
+            self.getHelp()
+            self.logger.info("DAO SQL queries to be executed:")
+            for sqlStr in self.getSqlQuery():
+                msg = "\n----------------------------------------------------------------------\n"
+                msg += "%s"
+                msg += "\n----------------------------------------------------------------------\n"
+                self.logger.info(msg, sqlStr)
+            self.logger.info("DAO SQL arguments provided:\n{}, {}".format(pformat(sqlArgs), pformat(sqlKwArgs)))
+        else:
+            results = self.dao.execute(*sqlArgs, **sqlKwArgs)
+            self.logger.info("DAO Results:\n{}".format(pformat(results if isinstance(results, dict) else list(results))))
+        return results
+
+    def getSqlQuery(self):
+        """
+        A simple method to inspect all DAO object attributes and accumulate any sql query it finds in a simple list
+        :return: A list of all sql queries it finds in the object.
+        """
+        sqlRegEx = re.compile(".*sql.*", re.IGNORECASE)
```

Comment:
there is not need to compile each time regexp. Instead, put it into constructor of the class.

Reply:
good point!